src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -259,23 +255,45 @@ private void SolutionEvents_AfterClosing()
             await InitializeAsync();
 
             var pendingRestore = _pendingRestore;
+            var shouldStartNewBGJobRunner = true;
+
+            // lock _pendingRequests to figure out if we need to start a new background job for restore
+            // or if there is already one running which will also take care of current request.
+            lock (_pendingRequests.Value)
```

Comment:
Use an `object` with a useful name instead of the lazy value here. Mostly for clarity, but also because I think Lazy does some allocations to return the value.